server/liveblog/client_modules/client_modules.py

```diff
@@ -557,6 +510,69 @@ def get_blog_posts(blog_id):
     return api_response(result_data, 200)
 
 
+@voting_blueprint.route("/api/client_poll_vote/<poll_id>", methods=["POST"])
+def client_poll_vote(poll_id):
+    """
+    Handles the voting action for a poll.
+
+    This function allows users to vote for a specific option in a poll. It performs the following steps:
+    1. Validate the request data
+    2. Retrieve the poll and validate selected option
+    3. Increment the vote count
+    4. Update associated blog posts to reflect change on embed
+
+    Returns:
+        JSON response with code:
+            - 201: If the vote is successfully recorded.
+            - 400: If the request is invalid or any check fails.
+            - 500: If an error occurs during vote update.
+
+    **TODO:**
+    * Add rate limiter decorator on endpoint
+    """
+    data = request.json
+    option_selected = data.get("option_selected", None)
+    if option_selected is None:
+        return api_error("Error: Option selected is required in request", 400)
+
+    polls = get_resource_service("polls")
+    poll = polls.find_one(req=None, _id=poll_id)
+    if poll is None:
+        return api_error("Error: Poll not found", 400)
+
+    poll_body = poll.get("poll_body", {})
+    answers = poll_body.get("answers", [])
+    option_exists = any(answer.get("option") == option_selected for answer in answers)
+    if not option_exists:
+        return api_error(
+            f"Error: Option '{option_selected}' not found in poll answers", 400
+        )
+
+    result = polls.find_and_modify(
+        query={"_id": poll_id, "poll_body.answers.option": option_selected},
+        update={"$inc": {"poll_body.answers.$.votes": 1}},
+        new=True,
+    )
+
+    if result is None:
+        return api_error("Error: Unable to update poll votes", 500)
+
+    blog_id = poll.get("blog")
+    for post in get_resource_service("client_posts").find(
+        {"blog": blog_id, "particular_type": "post"}
+    ):
+        for assoc in post_utils.get_associations(post):
+            if assoc.get("residRef") == poll_id:
+                updated_post = post.copy()
+                updated_post["content_updated_date"] = utcnow()
+                get_resource_service("posts").update(
+                    post.get("_id"), updated_post, post
+                )
+                app.blog_cache.invalidate(blog_id)
```

Comment:
following the single responsibility principle (I think you mentioned the other day), I'd suggest to put this into the posts service or in some util method instead, this way we avoid more tight coupling between these. What do you think?

Reply:
Sure. No problem